solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpJdkSolrClient.java

```diff
@@ -419,11 +425,7 @@ private synchronized boolean maybeTryHeadRequestSync(String url) {
   }
 
   private void decorateRequest(HttpRequest.Builder reqb, SolrRequest<?> solrRequest) {
-    if (requestTimeoutMillis > 0) {
-      reqb.timeout(Duration.of(requestTimeoutMillis, ChronoUnit.MILLIS));
-    } else if (idleTimeoutMillis > 0) {
-      reqb.timeout(Duration.of(idleTimeoutMillis, ChronoUnit.MILLIS));
-    }
+    reqb.timeout(Duration.of(requestTimeoutMillis, ChronoUnit.MILLIS));
```

Comment:
same comment as for Http2SolrClient: we now keep this simply and do defaulting logic in one common place earlier